### PR TITLE
Added wrapping starknet_py.TransactionFailedError in ProtostarException

### PR DESCRIPTION
**Pull request by Radinyn**

### CODE REVIEW

File: protostar/starknet_gateway/gateway_facade.py [hunk 1]
```diff
@@ -94,15 +95,18 @@ async def deploy(
...
+            register_response(dataclasses.asdict(result))
+            if wait_for_acceptance:
+                if self._logger:
+                    self._logger.info("Waiting for acceptance...")
+                _, status = await self._gateway_client.wait_for_tx(
+                    result.transaction_hash, wait_for_accept=wait_for_acceptance
+                )
+                result.code = status
+        except TransactionFailedError as ex:
+            raise TransactionException(str(ex)) from None
```
Comment (mkaput) [inline protostar/starknet_gateway/gateway_facade.py:109]:
```suggestion
            raise TransactionException(str(ex)) from ex
```

etc.

`from None` part has no sense to me and I would rather keep causality chain

**Review by mkaput — CHANGES_REQUESTED** [review]

File: protostar/starknet_gateway/gateway_facade.py (hunk 1 above)
Comment (Radinyn) [reply protostar/starknet_gateway/gateway_facade.py]: done

**Review by mkaput — APPROVED** [review]

File: protostar/starknet_gateway/gateway_facade.py [hunk 2]
```diff
@@ -168,15 +172,18 @@ async def declare(
...
+            register_response(dataclasses.asdict(result))
+            if wait_for_acceptance:
+                if self._logger:
+                    self._logger.info("Waiting for acceptance...")
+                _, status = await self._gateway_client.wait_for_tx(
+                    result.transaction_hash, wait_for_accept=wait_for_acceptance
+                )
+                result.code = status
+        except TransactionFailedError as ex:
+            raise TransactionException(str(ex)) from ex
```
Comment (Arcticae) [inline protostar/starknet_gateway/gateway_facade.py:186]: Shouldn't that be ex.message?

Comment (mkaput) [reply protostar/starknet_gateway/gateway_facade.py:186]: `str(ex)` decorates `ex.message` with information about exception kind

**Review by kasperski95 — APPROVED** [review]

**Pull request merged**